src/snowflake/snowpark/session.py

```diff
@@ -277,7 +278,13 @@ def __init__(self, conn: ServerConnection) -> None:
         self._file = FileOperation(self)
 
         self._analyzer = Analyzer(self)
-        self._sql_simplifier_enabled: bool = False
+        self._sql_simplifier_enabled: bool = (
+            conn._conn._session_parameters.get(
+                _PYTHON_SNOWPARK_USE_SQL_SIMPLIFIER_STRING, False
+            )
```

Comment:
@sfc-gh-mkeller Is this the right way to get a session parameter from a SnowflakeConnection if the parameter exists in the Snowflake database?

Reply:
You can use something similar to `self._use_scoped_temp_objects`

Reply:
It's aligned with use scoped temp objects

Reply:
Manually tested with my local regression environment. Confirmed that this works.